src/Microsoft.SqlTools.ServiceLayer/Rename/RenameUtils.cs

```diff
@@ -0,0 +1,74 @@
+        {
+            if (requestParams == null)
+            {
+                throw new ArgumentNullException();
+            }
+            if (String.IsNullOrEmpty(requestParams.TableInfo.TableName) || String.IsNullOrEmpty(requestParams.TableInfo.Database) || String.IsNullOrEmpty(requestParams.TableInfo.Schema) || String.IsNullOrEmpty(requestParams.TableInfo.OldName) || String.IsNullOrEmpty(requestParams.ChangeInfo.NewName))
+            {
+                throw new ArgumentException(SR.RenameRequestParametersNotNullOrEmpty);
+            }
+            if (!Regex.IsMatch(requestParams.TableInfo.Schema, RegexMatchPatternValidObjectName) || !Regex.IsMatch(requestParams.TableInfo.Database, RegexMatchPatternValidObjectName) || !Regex.IsMatch(requestParams.TableInfo.OldName, RegexMatchPatternValidObjectName) || !Regex.IsMatch(requestParams.TableInfo.TableName, RegexMatchPatternValidObjectName) || !Regex.IsMatch(requestParams.ChangeInfo.NewName, RegexMatchPatternValidObjectName))
+            {
+                throw new ArgumentOutOfRangeException(SR.NotAllowedCharacterForRenaming);
+            }
+        }
+        /// <summary>
+        /// Method to get the sqlobject, which should be renamed
+        /// </summary>
+        /// <param name="requestParams">parameters which are required for the rename operation</param>
+        /// <param name="connection">the sqlconnection on the server to search for the sqlobject</param>
+        /// <returns>the sqlobject if implements the interface IRenamable, so they can be renamed</returns>
+        public static IRenamable GetSQLRenameObject(ProcessRenameEditRequestParams requestParams, SqlConnection connection)
+        {
+            ServerConnection serverConnection = new ServerConnection(connection);
+            Server server = new Server(serverConnection);
+
+            Database database = new Database(server, requestParams.TableInfo.Database);
+            SqlSmoObject dbObject = database.Parent.GetSmoObject(new Urn(GetURNFromDatabaseSqlObjects(requestParams, serverConnection.TrueName)));
+
+            return (IRenamable)dbObject;
+        }
+        /// <summary>
+        /// Method to generate the Uniform Resource Name (URN) of a sqlobject 
+        /// </summary>
+        /// <param name="requestParams">parameters which are required for the rename operation</param>
+        /// <param name="trueNameOfServer">return the name of the server (@@Servername)</param>
+        /// <returns>URN string of the sqlobject</returns>
+        public static string GetURNFromDatabaseSqlObjects(ProcessRenameEditRequestParams requestParams, string trueNameOfServer)
+        {
+            String urnTableReplaced = String.Format(RenameUtils.UrnPrefix, trueNameOfServer, requestParams.TableInfo.Database, requestParams.TableInfo.TableName, requestParams.TableInfo.Schema, requestParams.TableInfo.OldName);
+
+            if (requestParams.ChangeInfo.Type == ChangeType.COLUMN)
```

Comment:
Instead of having to pass the type and build up the URN here I think it would be better just to do this on the ADS side, similar to what I'm doing for the properties window : https://github.com/microsoft/azuredatastudio/blob/main/extensions/admin-tool-ext-win/src/utils.ts#L129

(and what we could look into to make this even easier is to actually pass the URN into the tree node from STS in the first place. Then we wouldn't even have to manually construct it ourselves - and the URN would be useful for a lot of other stuff too. But that can be done separately from this PR - for now I think it'd be best to just assume that we'll be getting the URN from ADS already fully constructed)

Reply:
Thank you. I think you are right. Furthermore, I will change the properties in the class ```RenameRequestParams``` to that one:

```C#
public class RenameRequestParams
    {
        public string UrnOfObject{ get; set; }
        public string NewName { get; set; }
    }
```
I think that is more appropriate.